src/rpc/net.cpp

```diff
@@ -282,6 +283,61 @@ static UniValue addnode(const JSONRPCRequest& request)
     return NullUniValue;
 }
 
+static UniValue addconnection(const JSONRPCRequest& request)
+{
+    const RPCHelpMan help{"addconnection",
+        "\nOpen an outbound connection to a specified node (test only)",
+        {
+            {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "The IP address and port to attempt connecting to."},
+            {"connectiontype", RPCArg::Type::STR, RPCArg::Optional::NO, "'outbound' or 'blockrelay'"},
+        },
+        RPCResult{
+            RPCResult::Type::OBJ, "", "",
+            {
+                { RPCResult::Type::STR, "address", "Address of newly added connection." },
+                { RPCResult::Type::STR, "connectiontype", "Type of connection opened." },
+            }},
+        RPCExamples{
+            HelpExampleCli("addconnection", "\"192.168.0.6:8333\" \"outbound\"")
+            + HelpExampleRpc("addconnection", "\"192.168.0.6:8333\" \"outbound\"")
+        },
+    };
+
+    help.Check(request);
```

Comment:
nit: For new code it might be good to use the non-deprecated constructor (see e.g. #19528)

Reply:
done! how's it look now? 

q: is there anywhere that captures the current conventions around RPC functions? during original implementation I was looking around, but clearly wasn't able to successfully identify which methods are preferred